flink-yarn/src/main/java/org/apache/flink/yarn/ResourceInformationReflector.java

```diff
@@ -151,7 +151,9 @@ void setResourceInformationUnSafe(Object resource, String resourceName, long amo
 			for (int i = 2; i < externalResourcesInfo.length; i++) {
 				final String name = (String) resourceInformationGetNameMethod.invoke(externalResourcesInfo[i]);
 				final long value = (long) resourceInformationGetValueMethod.invoke(externalResourcesInfo[i]);
-				externalResources.put(name, value);
+				if (value > 0) {
+					externalResources.put(name, value);
+				}
```

Comment:
I think the proper solution is to enforce this invariant in the `InternalContainerResource` constructor (filtering out entries with a value of `0` or to adapt the `equals` method to not check entries with a value of `0`). Otherwise we might run into the situation that sometime in the future when the user can specify the GPU constraints on an operator and he sets it to `0` that the resources won't match.

Reply:
Yes, you are right.